src/main/java/org/reactome/release/dataexport/FTPFileUploader.java

```diff
@@ -0,0 +1,468 @@
+
+/**
+ * Class for updating Reactome export files on an FTP Server.  Classes which extend this abstract class will provide
+ * methods to:
+ *
+ * 1) Upload new (i.e. current Reactome release) files, created by classes in this project
+ * 2) Delete old (i.e previous Reactome release) files
+ * 3) List files present on the FTP Server for confirmation of successful file upload/deletion.
+ *
+ * @author jweiser
+ */
+public abstract class FTPFileUploader {
+	private static final Logger logger = LogManager.getLogger("mainLog");
+
+	private static final List<String> requiredProperties = Arrays.asList("outputDir", "reactomeVersion");
+
+	private Properties props;
+	private FTPClient ftpClientConnectionToServer;
+
+	/**
+	 * Creates an object for uploading new Reactome files (as well ad deleting old Reactome files) to an FTP Server.
+	 *
+	 * @param props The properties object with the key/value pairs providing connection and path information for
+	 * uploading to the FTP Server
+	 */
+	protected FTPFileUploader(Properties props) {
+		this.props = props;
+		throwIllegalStateExceptionUnlessRequiredPropsPresent();
+	}
+
+	/**
+	 * Returns the logger object for logging info, warnings, errors, etc. from activity in this class and sub-classes
+	 * methods.
+	 *
+	 * @return Logger object for logging class activity
+	 */
+	protected Logger getLogger() {
+		return logger;
+	}
+
+	/**
+	 * Initialization for the FTP Server does the following:
+	 *
+	 * 1) Connects to the FTP Server
+	 * 2) Enters local passive mode
+	 * 3) Logs into the FTP Server with the username and password from getUserName() and getPassword() methods,
+	 * respectively
+	 * 4) Changes to the working directory obtained from the getReactomeDirectoryPathOnFTPServer() method
+	 *
+	 * @throws IOException Thrown if unable, for the FTP Server, to connect, log in, or change working directory
+	 * @see #getUserName()
+	 * @see #getPassword()
+	 * @see #getReactomeDirectoryPathOnFTPServer()
+	 */
+	public void initializeFTPConnectionToServer() throws IOException {
+		getFtpClientToServer().connect(getServerHostName());
+
+		getFtpClientToServer().enterLocalPassiveMode();
+
+		if (loginToFTPServer()) {
+			logger.info("Login successful to " + getServerHostName());
+		} else {
+			logger.error("Login to " + getServerHostName() + " failed");
+		}
+
+		getFtpClientToServer().changeWorkingDirectory(getReactomeDirectoryPathOnFTPServer());
+	}
+
+	/**
+	 * Updates (uploads new files and deletes old files) on the FTP Server and logs the files which exist on the FTP
+	 * Server in the Reactome specific directory after the update is complete.
+	 *
+	 * @throws IOException Thrown if unable to upload new Reactome files to the FTP server, delete old Reactome files
+	 * from the FTP server, provide the listing of files on the FTP Server after the update is complete
+	 */
+	public void updateFilesOnServer() throws IOException {
+		if (!uploadFilesToServer()) {
+			return;
+		}
```

Comment:
This loses style points from me. 
How am I to read this? If the upload fails, then... exit the method early with a NULL return (from a void method, no less). 
Otherwise, delete files, log stuff, then close the FTP connection.

...and if you're going to keep it looking like this, why isn't the FTP connection closed before `return;`?

Reply:
The intention here was only to delete the older Reactome release files from the FTP server if all the new Reactome release files were successfully uploaded.  I had a bug in the Perl scripts that went ahead with the deletion of the old files even when the upload of the new files failed, so it left the FTP server without any Reactome content.

Fair point about the style of this - I'll rework it and also ensure the connection is closed regardless of the branch path the code takes.

Reply:
Okay, this has been redone if you'd like to take look: https://github.com/reactome/release-data-exporter/pull/4/commits/b8e643fe458d66a85fac4f8697f9180cc646c571